### ISSUE TITLE
Texture names with local characters

### ISSUE THREAD

**Issue by michaliskambi** [castle-engine/castle-engine#355]
Reporter notes on Discord https://discord.com/channels/389676745957310465/389676745957310467/923530888661319681:

> its a free chinese max 3d model, textures named in chinese, i use ru windows 10 currently
[...]
> exported with babylon from max 2018 directly
> mb its an issue of babylon, i m not 100% sure, but actually if i try to open the resulting gltf, view3d shows [] squares everywhere where > cyrillic or chinese should be

**Comment by michaliskambi** [comment]
[cadnav.com_model.zip](https://github.com/castle-engine/castle-engine/files/7769051/cadnav.com_model.zip)

**Comment by michaliskambi** [comment]
I looked into this. Confirm a bug, view3dscene cannot read textures referenced by this glTF file.

This is likely also a duplicate of our bug on https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/Unicode%E2%9D%A4%E2%99%BBTest .

The texture filenames are indeed in Chinese:

```
朱子柳.dds
朱子柳.png
朱子柳头发.dds
朱子柳头发.png
朱子柳透贴.dds
朱子柳透贴.png
```

They seem to be correctly referenced from glTF:

```
"images":[{"uri":"朱子柳.png"},{"uri":"朱子柳透贴.png"},{"uri":"朱子柳头发.png"}],
```

(Although the textures are in a subdirectory `Character_A1016A449`, not specified correctly in uri, but that's trivial to fix by moving them up. And still the issue remains.)

Debugging, it seems we do not get correct UTF-8 data from PasGLTF, below I output `URI` from `TPasGLTF.TImage` instance:

```
Using for image texture
æ±å­æ³.png
13
Using for image texture
æ±å­æ³éè´´.png
19
```

13 and 19 are the `UTF8Length` results. It's too much -- we should see only results 3+4=7 and 5+4 = 9. This shows we indeed have wrong data in string, looks like each Chinese character looks like 3 characters to us.

Tested on Linux with FPC 3.2.2 now.
